planner/core/preprocess.go

```diff
@@ -1592,11 +1592,14 @@ func (p *preprocessor) handleAsOfAndReadTS(node *ast.AsOfClause) {
 		return
 	}
 	if p.LastSnapshotTS != 0 {
-		dom := domain.GetDomain(p.ctx)
-		p.InfoSchema, p.err = dom.GetSnapshotInfoSchema(p.LastSnapshotTS)
+		is, err := p.ctx.GetSnapshotInfoSchema(p.LastSnapshotTS)
+		p.err = err
 		if p.err != nil {
 			return
 		}
+		if is != nil {
```

Comment:
`if is != nil` seems unnecessary. `if p.err != nil` can ensure infoschema not nil

Reply:
Rest LGTM